enhancements/insights/config-api.md

```diff
@@ -0,0 +1,296 @@
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	Gathering Gathering `json:"gathering,omitempty"`
+	// SimpleContentAcces spec attribute includes all the configuration options related to
+	// periodical downloading of the SCA certificates.
+	SimpleContentAccess SimpleContentAccess `json:"simpleContentAccess,omitempty"`
+	// ClusterTransfer spec attribute includes all the configuration options related to
+	// periodical check of pending cluster transfers.
+	ClusterTransfer ClusterTransfer `json:"clusterTransfer,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
+	// SimpleContentAccessStatus provides basic information about the SCA controller.
+	SimpleContentAccessStatus SimpleContentAccessStatus `json:"simpleContentAccessStatus,omitempty"`
+	// ClusterTransferStatus provides basic information about the ClusterTransfer controller.
+	ClusterTransferStatus ClusterTransferStatus `json:"clusterTransferStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"insightsReport,omitempty"`
+}
+
+type Gathering struct {
+	// Interval defines the time frequency of gathering. The duration is 2h by default.
+	// +kubebuilder:default=2h
+	Interval time.Duration `json:"interval,omitempty"`
+	// UploadEndpoint defines ingress endpoint for uploading Insights archive
+	// +kubebuilder:default=https://cloud.redhat.com/api/ingress/v1/upload
+	UploadEndpoint string `json:"endpoint,omitempty"`
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason
+	ForceGatheringReason string `json:"forceGatheringReason,omitempty"`
+	// List of gatherers to be included in the gathering. Empty list means that all gatherers are included
+	Gatherers []string `json:"gatherers,omitempty"`
+	// InsightsReport includes all the configuration options related to downloading the Insights report.
+	InsightsReport InsightsReport `json:"insightsReport,omitempty"`
+	// Flag to disable gathering. When the gathering is disabled then it cannot be forced with the `ForceGatheringReason`
+	Disabled bool `json:"disabled,omitempty"`
```

Comment:
this is weird placed :-) I can disable gathering by listing all gathers in Gathers field... can we just make an "All" option there and then we remove this field?

Reply:
Yes I think so. Good point thanks.

Reply:
I mentioned the "all" option in the `DisabledGatherers` field comment and removed this one.